### DISCUSSION TITLE
Central Istio Ingressgateway in a multi tenant cluster

### DISCUSSION THREAD

**Discussion by sambitr** [istio/istio#52377] [Q&A]
We use multi-tenant clusters with each tenant running in their own namespace. Everything is locked down with policies, RBAC ect. Currently we deploy an Ingress Gateway in each tenant namespace for them and they then control routing to their pods/services using Virtual Services. There is an app gateway running outside of the clusters which routes traffic to the appropriate tenant ingress using path based routing.

This works fine however it is wasteful having so many ingress gateways. Is there a way that we can create a central Ingress Gateway which routes to the appropriate tenant namespace using path based routing? We would still need the tenants to do additional path based routing within their namespace.

E.g.
Central Ingress gateway /app1/* => some service in app1 namespace
/app1/microservice1 => microservice1 pod/svc in app1 namespace
/app1/microservice2 => microservice2 pod/svc in app1 namespace

We would need to ensure that app2 couldn’t route traffic intended for app1 to their namespace.

**Comment by abasitt** [comment]
I have yet to explore kubernetes gateway-api more but this seems to be the good use-case for that and how you can allow a gateway resource to attach HTTP route from specific namespaces. You will have centralize control over deployment and gateways.
If you want to stick with VS, i think delegate feature may solve your problem too. https://istio.io/latest/docs/reference/config/networking/virtual-service/#Delegate

**Reply by sambitr** [reply]
Hey @abasitt , thanks for this suggestion. It solves the first purpose of central ingress and routing traffic. But how about the second problem: ensure that app2 couldn’t route traffic intended for app1 to their namespace?

**Reply by Babbili** [reply]
you may use [network policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) for that
